lib/circuitry/config/subscriber_settings.rb

```diff
@@ -0,0 +1,32 @@
+require 'virtus'
+require 'circuitry/config/shared_settings'
+
+module Circuitry
+  module Config
+    class SubscriberSettings
+      include Virtus::Model
+      include SharedSettings
+
+      attribute :queue_name, String
+      attribute :dead_letter_queue_name, String
+      attribute :visibility_timeout, Integer, default: 30 * 60
+      attribute :max_receive_count, Integer, default: 8
+      attribute :lock_strategy, Object, default: ->(_page, _att) { Circuitry::Locks::Memory.new }
+
+      def dead_letter_queue_name
+        super || "#{queue_name}-failures"
+      end
+
+      def async_strategy=(value)
+        validate_setting(value, Subscriber.async_strategies)
+        super
+      end
+
+      def lock_strategy=(value)
+        unless [:soft_lock, :have_lock, :unlock].all? { |s| value.respond_to?(s) }
```

Comment:
Yeah, that would work.  I thought we had that, but maybe it's something I meant to add and never did.